project/src/main/scala/licenses/DependencySummary.scala

```diff
@@ -122,37 +123,47 @@ object ReviewedSummary {
 
   /** Returns a list of warnings that indicate missing reviews or other issues.
     */
-  def warnAboutMissingReviews(summary: ReviewedSummary): WithWarnings[Unit] = {
-    val warnings = summary.dependencies.flatMap { dep =>
-      val warnings = collection.mutable.Buffer[String]()
-      val name     = dep.information.moduleInfo.toString
+  def warnAboutMissingReviews(
+    summary: ReviewedSummary
+  ): WithDiagnostics[Unit] = {
+    val diagnostics = summary.dependencies.flatMap { dep =>
+      val diagnostics = collection.mutable.Buffer[Diagnostic]()
+      val name        = dep.information.moduleInfo.toString
 
       val missingFiles = dep.files.filter(_._2 == AttachmentStatus.NotReviewed)
       if (missingFiles.nonEmpty) {
-        warnings.append(
-          s"${missingFiles.size} files are not reviewed in $name."
+        diagnostics.append(
+          Diagnostic.Problem(
+            s"${missingFiles.size} files are not reviewed in $name."
+          )
         )
       }
       val missingCopyrights =
         dep.copyrights.filter(_._2 == AttachmentStatus.NotReviewed)
       if (missingCopyrights.nonEmpty) {
-        warnings.append(
-          s"${missingCopyrights.size} copyrights are not reviewed in $name."
+        diagnostics.append(
+          Diagnostic.Problem(
+            s"${missingCopyrights.size} copyrights are not reviewed in $name."
+          )
         )
       }
 
       val includedInfos =
         (dep.files.map(_._2) ++ dep.copyrights.map(_._2)).filter(_.included)
       if (includedInfos.isEmpty) {
-        warnings.append(
-          s"No files or copyright information are included for $name."
+        diagnostics.append(
+          Diagnostic.Problem(
+            s"No files or copyright information are included for $name. Generally every dependency should have _some_ copyright info, so this suggests all our heuristics failed. Please find the information manually and add it using `files-add` or `copyright-add`. Even if the dependency is public domain, it may be good to include some information about its source."
```

Comment:
```suggestion
            s"No files or copyright information are included for $name. Generally every dependency should have _some_ copyright info, so this suggests all our heuristics failed. Please find the information manually and add it using `files-add` or `copyright-add`. Even if the dependency is in public domain, it may be good to include some information about its source."
```